node.go

```diff
@@ -235,13 +235,56 @@ func fieldRepetitionTypeOf(node Node) format.FieldRepetitionType {
 	}
 }
 
-type Group map[string]Node
+type Group struct {
+	nodes    map[string]Node
+	optional bool
+}
```

Comment:
Instead of changing `Group`, would it work if we used the `Optional` wrapper?
```go
Optional(Group{ ... })
```

Reply:
mmm will take a look but we are gonna have a type issues with `mapNode` which is expecting a Group and not a Node right ?

Reply:
Actually we can vary the tag there. Thanks ! Will refactor :)